core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/html.kt

```diff
@@ -196,6 +204,36 @@ internal fun AnyFrame.toHtmlData(
     return DataFrameHtmlData(style = "", body = body, script = script)
 }
 
+private interface DataFrameLike {
+    fun configuration(default: DisplayConfiguration): DisplayConfiguration
+
+    fun df(): AnyFrame
+}
+
+private fun dataFrameLikeOrNull(value: Any?): DataFrameLike? =
```

Comment:
I'd probably make this an extension function, like `Any?.asDataFrameLikeOrNull()`, similar to `.asIntOrNull()` etc.